File: Week6-Python/pset6/readability/readability.py
from cs50 import get_string


def main():
    text = get_string("Text: ")
    letter_cn = letters(text)
    words_cn = words(text)
    sentences_cn = sentences(text)
    L = (100 / words_cn) * letter_cn
    S = (100 / words_cn) * sentences_cn
    index = round(0.0588 * L - 0.296 * S - 15.8)
    if index >= 16:
        print("Grade 16+")
    elif index < 1:
        print("Before Grade 1")
    else:
        print("Grade:", index)


def letters(text):
    letter_cn = 0
    for i in text:
        if i.isalpha() == True:
            letter_cn = letter_cn + 1
    return letter_cn


def words(text):
    word_cn = 1  # need to count first word of text which is not preceded by " "
    for i in text:
        if i == " ":
            word_cn = word_cn + 1
    return word_cn


def sentences(text):
    sentence_cn = 0
    for i in text:
        if i == "." or i == "!" or i == "?":
            sentence_cn = sentence_cn + 1
    return sentence_cn


main()

File: Week2-Arrays/pset2/readability.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

// step 5, calculate and return coleman-liau, whole number
// if output is less than "Before Grade 1"
// "Grade 16+"

int letters(string text);
int words(string text);
int sentences(string text);

int main(void)
{
    string text = get_string("Text: ");
    float letter_cn = letters(text);
    float words_cn = words(text);
    float sentences_cn = sentences(text);
    float L = (100 / words_cn) * letter_cn;
    float S = (100 / words_cn) * sentences_cn;
    int index = round(0.0588 * L - 0.296 * S - 15.8);
    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

int letters(string text)
{
    int letter_cn = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        text[i] = (char) text[i];
        if (isalpha(text[i]) != 0)
        {
            letter_cn = letter_cn + 1;
        }
    }
    return letter_cn;
}

int words(string text)
{
    int word_cn = 1; // need to count first word of text which is not preceded by " "
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == ' ')
        {
            word_cn = word_cn + 1;
        }
    }
    return word_cn;
}

int sentences(string text)
{
    int sentence_cn = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentence_cn = sentence_cn + 1;
        }
    }
    return sentence_cn;
}